typescript/src/v0/Mcap0RecordWriter.ts

```diff
@@ -0,0 +1,90 @@
+import Writer from "../common/Writer";
+import { IWritable } from "./IWritable";
+import { MCAP0_MAGIC, Opcode } from "./constants";
+import { ChannelInfo, Header, Footer, Message, Attachment } from "./types";
+
+/**
+ * Mcap0RecordWriter provides methods to serialize mcap records to an IWritable.
+ *
+ * It makes no effort to ensure spec compatability on the order of records, this is the responsibility
+ * of the caller.
+ *
+ * Unless you are building your own higher level writer interface, you'll likely want to use one of
+ * the higher level writer interfaces.
+ */
+export class Mcap0RecordWriter {
+  private writable: IWritable;
+
+  constructor(writable: IWritable) {
+    this.writable = writable;
+  }
+
+  async writeMagic(): Promise<void> {
+    await this.writable.write(new Uint8Array(MCAP0_MAGIC));
+  }
+
+  async writeHeader(header: Header): Promise<void> {
+    const serializer = new Writer();
```

Comment:
```suggestion
    const writer = new Writer();
```

Reply:
I am not a fan of the class name Writer. In your PR I commented this should be called DatatypeWriter, DatatypeSerializer something other than "writer".

Reply:
Ok, I don't think "datatype" makes it more clear, but open to other renames. The variable name should match the type if possible though.